src/AbstractFactory/ReflectionBasedAbstractFactory.php

```diff
@@ -249,4 +250,30 @@ private function resolveParameter(ReflectionParameter $parameter, ContainerInter
         // default defined.
         return $parameter->getDefaultValue();
     }
+
+        /**
+     * Gets the parameter's class name.
+     *
+     * @param ReflectionParameter $parameter
+     *   The parameter.
+     *
+     * @return string|null
+     *   The parameter's class name or NULL if the parameter is not a class.
+     */
+    public static function getParameterClassName(ReflectionParameter $parameter)
```

Comment:
This method is declared as static, but is never used in this form. And I see no reason to extend the public API here.
Please set the visibility to private and remove `static`.